src/plugins/discover/public/application/components/top_nav/get_top_nav_links.tsx

```diff
@@ -305,6 +305,7 @@ export const getTopNavLinks = (
       defaultMessage: 'New',
     }),
     run() {
+      query.filterManager.setFilters([]); // resetting the filters while we are loading a new search
```

Comment:
Lets add a test for this as well.

Reply:
Added additional tests in https://github.com/opensearch-project/opensearch-dashboards-functional-test/pull/1585